### PR TITLE
Linear regression

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

File: bigml/linear.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+            input_array.extend(new_inputs)
+
+        if self.bias or not self.compact:
+            input_array.append(1)
+
+        return input_array
+
+    def categorical_encoding(self, inputs, field_id, compact):
+        """Returns the prediction and the confidence intervals
```
Comment (jaor) [inline bigml/linear.py]: is this docstring correct? the name of the method seems weird if so...

File: docs/index.rst [hunk 2]
```diff
@@ -1808,6 +1809,174 @@ logistic function as well as the configuration parameters described in
 the `developers section <https://bigml.com/api/logisticregressions>`_ .
 
 
+Linear Regressions
+------------------
+
+A linear regression is a supervised machine learning method for
+solving regression problems. The implementation is a multiple linear regression
```
Comment (jaor) [inline docs/index.rst]: hmm, saying that the implementation of a linear regression is a multiple linear regression sound a bit circular :) perhaps a "multi-linear approximation"?

File: docs/index.rst [hunk 3]
```diff
@@ -4319,6 +4492,37 @@ be the maximum number of iterations:
     ensemble = api.create_ensemble('dataset/5143a51a37203f2cf7000972', args)
 
 
+Creating linear regressions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For regression problems, you can choose also linear regressions to model
+your data. Linear regressions expect the predicted value for the objective
+field to be computable as a linear combination of the predictions.
```
Comment (jaor) [inline docs/index.rst]: predictors

File: bigml/linear.py (hunk 1 above)
Comment (mmerce) [reply bigml/linear.py]: it isn't... cut and paste strikes again

File: docs/index.rst (hunk 2 above)
Comment (mmerce) [reply docs/index.rst]: yes, probably, but that's what the API docs say:

A linear regression is a supervised machine learning method that implements multiple linear regression for solving regression problems.

**Pull request merged**